### ISSUE TITLE
mint-ui: Unexpected behavior from Ui.Button 

### ISSUE THREAD

**Issue by AmitPress** [mint-lang/mint#363]
was trying to use ui components from mint ui...
the library and its dependency downloaded successfully...
but after trying to use one of the component(Button), it is now showing error all the time whether I am using the component or not.

Error Message: "**You are trying to reference an other entity in a top level entity before it is initialized**."

I often found this error while doing this 

property initval : Number = 0
state val = initval

I also tried to do so use componentOnMount lifecycle method but was unsuccessful.

**Comment by AmitPress** [comment]
``` 
property initval : Number = 0
state val = initval 
``` 
This code is working with componentDidMount.

**The code didn't worked then:**
Error Message: You are trying to reference an other entity in a top level entity before it is initialized.

```
component Child{
    property nineteen = 19  /* The entity you are referencing from: */
    state count = nineteen  /* Then entity you are referencing */
    
    fun up {
        next { count = count + 1 }
    }
    fun reset {
        next { count = 0}
    }

    fun render {
        <>
        <button onClick={up}>
            <{ Number.toString(count) }>
        </button>
        <button onClick={reset}>
            <{ "Reset" }>
        </button>
        </>
    }
}
```
The code worked:
```
component Child{
    property nineteen = 19  
    state count = 0  
    fun componentDidMount {
        next {
            count = nineteen
        }
    }
    fun up {
        next { count = count + 1 }
    }
    fun reset {
        next { count = 0}
    }

    fun render {
        <>
        <button onClick={up}>
            <{ Number.toString(count) }>
        </button>
        <button onClick={reset}>
            <{ "Reset" }>
        </button>
        </>
    }
}
```

**Comment by gdotdesign** [comment]
This is fixed in `0.12.0`.
